### DISCUSSION TITLE
Is there a way to create a corresponding .data file alongside your .yaml/.json file?

### DISCUSSION THREAD

**Discussion by 8BitSensei** [decaporg/decap-cms#7226] [Q&A] [answered]
I'm planning to use Decap to manage a data set which is using the [Open Data Institute template](https://github.com/theodi/data-publishing-template?tab=readme-ov-file). It requires me to create alongside every file containing data a corresponding .data file describing metadata, so for example you would have `data\my-data.json` and `data\my-data.data`. The metadata looks like it's just a YAML front matter with some basic information that can either be duplicated from the data or left as default:
```
---
category: data
filename: May_2014.csv
weight: 1
title: May 2014
description: Transactions in May 2014
type: text/csv
---
```
Is there anyway Decap CMS can automatically create this when ever I publish a file?

**Comment by martinjagodic** [comment, marked as answer]
Maybe you can create a [custom widget](https://decapcms.org/docs/custom-widgets/) that can save a file?

**Reply by 8BitSensei** [reply]
Yes, that looks like the way forward if nothing already exists. My other option is just to write a script that does this and make it a part of the deployment action, but I'd rather keep the solution part of Decap.
